src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
> id of diagnostic is opaque. can't force it to be unique across workspace.

Ah, so that means `_idToProviderAndTagger` map already has workspace separation in its key. The issue was just that we were just processing removal from subject buffer's previously owned workspace. The current change seems 👍 then. I agree with @CyrusNajmabadi that consuming these aggregated diagnostic updated events is not trivial.